src/pages/Templates.svelte

```diff
@@ -130,40 +134,48 @@
                     <input type="text" class="form-control" bind:value={search} on:input={filterTemplates} />
                 </div>
             </div>
-            <div class="row no-gutters">
+            <div class="row no-gutters" style="margin-top: 10px;">
                 {#if templates}
-                {#each filteredTemplates.filter(t => justMyTemplates ? t.userId === userdata.id : true) as template}
+                {#each filteredTemplates.filter(t => justMyTemplates ? t.userId === userdata.id : true) as template, i}
                 <div class="col-12 col-md-4" style="position:relative;">
-                    <div class="floating-card mdc-elevation--z20">
-                        <div class="row no-gutters">
+                    <div class="floating-card mdc-elevation--z8">
+                        <div class="row no-gutters" style="margin-left: 8px; margin-top: 4px;">
                             <div class="col-12 no-gutters">
                                 <h5>{template.name}</h5>
                             </div>
                         </div>
                         <button type="btn btn-primary" class="template-choice" on:click={() => clickTemplate(template.id)}>
-                            <img src={template.url} alt="template" style="width:100%;" />
-                            <div class="row">
-                                <div style="text-align:left" class="col">
-                                    <input class="color-display" type="color" readonly value={template.color_friendcode} />
-                                    Friend Code
+                            <div class="card-preview">
+                                {#if i === 0}
+                                <div bind:clientWidth={templateWidth}>
```

Comment:
What's the purpose of this line?

Reply:
This binds the width of the first template image to a variable. We use the clientWidth of the div element to determine how large to make the font when the page resizes. We cannot use the clientWidth of the img due to some technical limitation: my reference is https://svelte.dev/tutorial/dimensions. `{#if i === 0}` limits this to the first template image. That way we only measure one image and can resize all of the preview text on the page. This assumes that all the template previews in the page are the same size.

Reply:
I just realized that this can cause a bug if the user deletes the first template that is shown, since user templates are shown before everyone else's.

Reply:
The page appears to reload after deleting a template, so not sure how I got the bug to manifest in the first place.

Reply:
Perhaps you got a bug when resizing the page?